e2e/support/helpers/e2e-snowplow-helpers.js

```diff
@@ -40,6 +41,41 @@ export const expectGoodSnowplowEvents = count => {
     .should("have.length", count);
 };
 
+export const expectGoodSnowplowEventCount = expectedEventCounts => {
```

Comment:
this is super experimental and want to get y'all's opinion on whether this makes sense or not. I think it's better to be explicit about the events that we're asserting on (since in search, we have `new_search_query` and `search_results_filtered`) and I wanted a way to differentiate between the two

Reply:
The only thing I'm worried about is that chain of `event.event?.unstruct_event?.data?.data` - I'm trying to figure out if this will be a reliable check

Reply:
Oh ok that's perfect - I obviously couldn't find it but I'd rather use something that's already there than something new

Reply:
I also didn't notice that function, I will use it in my pr thanks.
One small thing i noticed though is that if we pass the wrong event name the e2e just fails with "Snowplow retry timeout" which I'm afraid will be confused with a flake instead of an actual failed test

Reply:
> i noticed though is that if we pass the wrong event name the e2e just fails with "Snowplow retry timeout" which I'm afraid will be confused with a flake instead of an actual failed test

Agree, the error message could have been better. It's not in the scope of this PR, but I wouldn't mind if you want to make that happen 😄